debezium-operator-core/src/main/java/io/debezium/operator/core/dependent/RoleDependent.java

```diff
@@ -23,16 +28,45 @@ public RoleDependent() {
 
     @Override
     protected Role desired(DebeziumServer primary, Context<DebeziumServer> context) {
+
+        List<PolicyRule> policyRules = new ArrayList<>();
+
+        policyRules.add(new PolicyRuleBuilder()
+                .withApiGroups("")
+                .withResources("secrets")
+                .withVerbs("get", "list", "watch")
+                .build());
+
+        policyRules.add(new PolicyRuleBuilder()
+                .withApiGroups("")
+                .withResources("configmaps")
+                .withVerbs(List.of("get", "list", "watch"))
+                .build());
+
+        if (primary.getSpec().getSource().getOffset().getActiveStore() instanceof ConfigMapOffsetStore) {
+            policyRules.add(getOffsetConfgiMapPolicyRule(primary));
+        }
+
         return new RoleBuilder()
                 .withMetadata(new ObjectMetaBuilder()
                         .withName(ROLE_NAME.formatted(primary.getMetadata().getName()))
                         .withNamespace(primary.getMetadata().getNamespace())
                         .build())
-                .withRules(new PolicyRuleBuilder()
-                        .withApiGroups("")
-                        .withResources("secrets", "configmaps")
-                        .withVerbs("get", "list", "watch")
-                        .build())
+                .withRules(policyRules)
                 .build();
     }
+
+    private static PolicyRule getOffsetConfgiMapPolicyRule(DebeziumServer primary) {
+
+        String name = primary.getSpec().getSource().getOffset().getConfigMap().getName() != null ? primary.getSpec().getSource().getOffset().getConfigMap().getName()
+                : OffsetsConfigMapDependent.managedName(primary);
```

Comment:
Make this more readable -- personally I like to split such long method call chains. 

E.g. passing the offset or even configMap model instead of the entire DebeziumServer to this method.